2023-risks/m1-insecure-authentication-authorization

```diff
@@ -0,0 +1,124 @@
+
+Moreover, failures in authentication can also expose underlying authorization failures. When authentication controls fail, the solution is unable to verify the user's identity, which is closely tied to a user's role and associated permissions. If an attacker can anonymously execute sensitive functionality, it indicates that the underlying code is not verifying the permissions of the user issuing the request for the action. Consequently, the anonymous execution of code underscores failures in both authentication and authorization controls.
+
+# Business Impacts
+
+**Application / Business Specific**
+
+The business impact of poor authentication and authorization will typically result in the following at a minimum:
+
+- Reputation Damage;
+- Information Theft;
+- Fraud;
+- Unauthorized Access to Data.
+
+# Am I Vulnerable To 'Insecure Authentication / Authorization'?
+
+Understanding the difference between authentication and authorization is paramount in evaluating mobile application security. Authentication identifies an individual, while authorization verifies if the identified individual has the necessary permissions for a particular action. These two aspects are closely related, as authorization checks should immediately follow mobile device request authentication.
+
+Insecure authorization can occur when an organization fails to authenticate an individual before executing a requested API endpoint from a mobile device, as it is virtually impossible to conduct authorization checks on an incoming request without an established caller's identity.
+
+Here are some straightforward indicators of insecure authorization:
+
+* __Presence of Insecure Direct Object Reference (IDOR) vulnerabilities__ - Noticing an IDOR vulnerability may suggest that the code isn't conducting a proper authorization check.
+* __Hidden Endpoints__ - Developers might neglect authorization checks on backend hidden functionality, assuming that the hidden functionality will only be accessed by a user with the appropriate role.
+* __User Role or Permission Transmissions__ - Should the mobile app transmit the user's roles or permissions to a backend system as part of a request, this could signal insecure authorization.
+
+Similarly, mobile apps can exhibit various signs of insecure authentication:
+
+* __Anonymous Backend API Execution__ - The ability of the app to execute a backend API service request without providing an access token may point to insecure authentication.
+* __Local Storage of Passwords or Shared Secrets__ - If the app stores any passwords or shared secrets locally on the device, this could be a sign of insecure authentication.
+* __Weak Password Policy__ - The use of a simplified password-entering process may imply insecure authentication.
+* __ Usage of Features like FaceID and TouchID__ - Employing features like FaceID or TouchID could be indicative of insecure authentication.
```

Comment:
using FaceID and TouchID only to unlock the TPM for accessing secrets that are one factor in authentication should be means for preventing insecure authentication and be suggested in the next section